modules/market/market.models.ts

```diff
@@ -8,3 +8,113 @@ export const Market = mongo.createModel<Types.MarketDocument>('Market', {
+  categoryGoals: [
+    {
+      category: { type: String, required: true },
+      goalPercentage: { type: Number, required: true },
+      currentPnL: { type: Number },
+      historicalPnL: { type: Number },
+    },
+  ],
+  investments: { type: mongo.Schema.Types.ObjectId, required: true },
+  totalPnL: { type: Number },
+});
+
+export const Investment = mongo.createModel<Types.InvestmentDocument>('Investment', {
+  type: { type: String, required: true },
+  amount: { type: Number, required: true },
+  category: { type: String, required: true },
+  purchasePrice: { type: Number, required: true },
+  currentValue: { type: Number, required: true },
+  pnl: { type: Number, required: true },
+  transactions: [
+    {
+      date: { type: Date, required: true },
+      type: { type: String, required: true },
+      price: { type: Number, required: true },
+      quantity: { type: Number, required: true },
+    },
+  ],
+  stock: { type: mongo.Schema.Types.ObjectId },
+  chainToken: { type: mongo.Schema.Types.ObjectId },
+});
+
+export const Stock = mongo.createModel<Types.StockDocument>('Stocks', {
+  ticker: { type: String, required: true },
+  companyName: { type: String, required: true },
+  companyId: { type: mongo.Schema.Types.ObjectId, required: true, ref: 'StockCompany' },
+  currentPrice: { type: Number, required: true },
+  marketCap: { type: Number },
+  volume: { type: Number },
+  currency: { type: String, required: true },
+  AUM: { type: Number },
+  quoteType: { type: String },
+  dailyChange: { type: Number },
+  dailyChangePercent: { type: Number },
+});
+
+export const ChainToken = mongo.createModel<Types.ChainDocument>('ChainToken', {
+  name: { type: String, required: true },
+  symbol: { type: String, required: true },
+  currentPrice: { type: Number, required: true },
+});
+
+export const StockCompany = mongo.createModel<Types.StockCompanyDocument>('StockCompany', {
+  name: { type: String, required: true },
+  ticker: { type: String, required: true },
+  country: { type: String, required: true },
+  industry: { type: String, required: true },
+  sector: { type: String, required: true },
+  type: { type: String, required: true },
+  issuedETFs: [{ type: mongo.Schema.Types.ObjectId, ref: 'ETF' }],
+});
+
+export const ETF = mongo.createModel<Types.ETFDocument>('ETF', {
+  name: { type: String, required: true },
+  ticker: { type: String, required: true },
+  issuer: { type: mongo.Schema.Types.ObjectId, required: true },
+  leverage: { type: Number, required: false },
+});
+
+export const StockSentiment = mongo.createModel<Types.StockSentimentDocument>('StockSentiment', {
+  ticker: { type: String, required: true },
+  sentiment: {
+    label: { type: String, required: true },
+    score: { type: Number, required: true },
+  },
+  context: { type: String, required: true },
+  confidence: { type: Number, required: true },
+});
+
+export const BigData = mongo.createModel<Types.BigDataDocument>('BigData', {
```

Comment:
use the Data model from Core?


// Data Model
export const Data = mongo.createModel<Types.DataDocument>(
  'Data',
  {
    mod: { type: String, required: true },
  },
  {
    virtuals: [...addTagVirtuals('Data'), ...addApplicationVirtual()],
  }
);